packages/api-explorer/src/JsonForm/anyOf-custom-editor.js

```diff
@@ -0,0 +1,206 @@
+      e.stopPropagation();
+
+      const selectedValues = Array.from(e.currentTarget.selectedOptions).map(selectedValue => selectedValue.value)
+      if (selectedValues.length === 0) {
+        self.showErrorMessage('empty-selection')
+      } else if (selectedValues.length === 1) {
+        self.hideErrorMessage()
+        self.switchEditor(self.display_text.indexOf(selectedValues[0]))
+      } else {
+        self.updateEditor(selectedValues)
+      }
+      self.onChange(true)
+      choices.hideDropdown()
+    });
+
+    /**
+     * FormSubmission event is dispatched by APIExplorer Doc component on formsubmission for
+     * listener to apply custom logics on the event.
+     */
+    switcher.onFormSubmission = () => {
+      const {selectedOptions} = switcher
+      if (!selectedOptions || selectedOptions.length === 0) {
+        self.showErrorMessage('empty-selection')
+        throw new Error('invalid form, user must select a schema')
+      }
+    }
+
+    return { switcher, switcherDiv }
+  },
+  updateEditor(selectedValues) {
+    const joinedValues = selectedValues.join('_')
+    const mergedEditorIndex = editorIndexesMap.get(joinedValues)
+    if (mergedEditorIndex) {
+      this.hideErrorMessage()
+      this.switchEditor(mergedEditorIndex);
+      return
+    }
+    const schemasIndexes = selectedValues.map(value => this.display_text.indexOf(value))
+    const schemas = this.types.filter((_, index) => schemasIndexes.includes(index))
+    const mergedSchemas = mergeSchemas(schemas, self)
+    if (!mergedSchemas) {
+      this.showErrorMessage('not-compatible')
+      return
+    }
+    this.hideErrorMessage()
+    const index = this.types.length
+    this.insertNewEditor(index, joinedValues, mergedSchemas)
+    this.switchEditor(index)
+  },
+  addErrorMessageHtmlNode() {
+    this.errorMessageHtmlNode = document.createElement('p')
+    this.errorMessageHtmlNode.style.color = 'red'
+    this.errorMessageHtmlNode.style.fontWeight = 'bold'
+    this.errorMessageHtmlNode.style.display = 'none'
+    this.errorMessageHtmlNode.style.marginTop = '10px'
+    this.container.appendChild(this.errorMessageHtmlNode)
+  },
+  showErrorMessage(type) {
+    this.hideEditor()
+    const errorsByType = {
+      'not-compatible': 'The selected schemas are not compatible!',
+      'empty-selection': 'Select at least 1 schema',
+    }
+    this.errorMessageHtmlNode.innerText = errorsByType[type] || 'Unknown error!'
+    this.errorMessageHtmlNode.style.display = 'block'
+  },
+  hideErrorMessage() {
+    this.editor_holder.style.display = 'block'
+    this.errorMessageHtmlNode.style.display = 'none'
+  },
+  insertNewEditor(index, joinedValues, mergedSchemas) {
+    editorIndexesMap.set(joinedValues, index)
+    this.types.push(mergedSchemas)
+    this.editors.push(false)
+  },
+  hideEditor() {
+    this.editor_holder.style.display = 'none'
+  }
+})
+
+const editorIndexesMap = new Map()
+
+const extractSchemasTypeIfCompatible = (schemas) => {
```

Comment:
@silversoul93 can you please check this function implementation? I think that now it's far more comprehensible but I'd like to know if it's enough or am I missing something?

Reply:
I think that it's ok